### PR TITLE
apply the same USB descriptor's iSerialNumber logic for ESP32-S2 as t…

### PR DESCRIPTION
**Pull request by lyusupov**
## Description of Change

ESP32-S**3** is currently using certain method to assign a default USB _iSerialNumber_ field for USB descriptor by specifying "_ _MAC_ _" keyword.
This PR applies the same method to the ESP32-S**2** target as well.

### CODE REVIEW

File: cores/esp32/USB.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@
 #define USB_PRODUCT ARDUINO_BOARD
 #endif
 #ifndef USB_SERIAL
-#if CONFIG_IDF_TARGET_ESP32S3
+#if CONFIG_IDF_TARGET_ESP32S2 || CONFIG_IDF_TARGET_ESP32S3
```
Comment (me-no-dev) [inline cores/esp32/USB.cpp:37]: The USB bootloader inside ESP32-S2 uses Serial "0". This is here for that exact reason. Please revert this change and set it manually if you need something specific.

Comment (lyusupov) [reply cores/esp32/USB.cpp]: done in https://github.com/espressif/arduino-esp32/pull/7563/commits/537dc7ce99a6db7e2528cd85f9f7951ec22ea995
